### PR TITLE
Added passing login credentials from EI Frontend to EI Backend

### PR DESCRIPTION
**Pull request by ValentinTyhonov**
Checking if username and password fields are not empty before logging
Sending GET request to backend for authorization
Presenting username after authorization and "Guest" in other cases
Hiding and showing login and logout buttons when it is needed
Removed registration and forgot password pages
Moved login button to top-right
Changed HttpClient and Response to CloseableHttpClient and CloseableResponse
Using single httpClient for all requests per session
Using localStorage for storing currentUser while user is logged in

### CODE REVIEW

File: src/main/resources/static/js/login.js [hunk 1]
```diff
@@ -1,13 +1,81 @@
...
+        });
+    }
+    // /Stop ## Global AJAX Sender function ##################################
+
+    // /Start ## Knockout ####################################################
+    function loginModel() {
+        this.userState = {
+            username: ko.observable(""),
+            password: ko.observable(""),
+            remember: ko.observable(false)
```
Comment (xmairaa) [inline src/main/resources/static/js/login.js]: i did not found **remember** related functionality.

Comment (xmairaa) [reply src/main/resources/static/js/login.js]: When user check this checkbox, the user name and password will store in cookies,
If user tries to login the cookies details are automatically populates into UserName and Password field.

Is there any other functionality for this remember? what is the use to store this checkbox value?

Comment (ValentinTyhonov) [reply src/main/resources/static/js/login.js]: oh. as I understand now, it is not needed to pass remember value to backend, but add this functionality to frontend.. I will fix it. thanks

File: src/main/resources/static/js/login.js [hunk 2]
```diff
@@ -1,13 +1,110 @@
...
-	
+
+    // /Start ## Global AJAX Sender function ##################################
+    var AjaxHttpSender = function () {};
+
+    AjaxHttpSender.prototype.sendAjax = function (url, type, data, callback) {
+        $.ajax({
+            url : url,
+            type : type,
+            data : data,
```
Comment (Christoffer-Cortes) [inline src/main/resources/static/js/login.js]: The way spring security works by default is to look for an Authorization header.
Attaching such a header would look like this:
beforeSend: function (xhr) {
    xhr.setRequestHeader ("Authorization", "Basic " + btoa(username + ":" + password));
},
btoa being a function that encodes the user info to base64.

**Comment by xmairaa** [comment]
Why Guest is fixed here, we can show logged username

**Comment by xmairaa** [comment]
looks good

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java [hunk 3]
```diff
@@ -180,39 +162,11 @@ public String getEIBackendSubscriptionAddress() {
         LOG.debug("Input Request JSON Content to be forwarded:\n" + inputReqJsonContent);
         HttpEntity inputReqJsonEntity = new ByteArrayEntity(inputReqJsonContent.getBytes());
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Why removing the HTTP Return code from debug Message? Good to know when debugging the web-server.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java [hunk 4]
```diff
@@ -120,38 +131,9 @@ public String getEIBackendSubscriptionAddress() {
...
-            }
-            bufReader.close();
-            inStream.close();
-        } catch (IOException e) {
-            LOG.error("Forward Request Errors: " + e);
-        }
-
-        LOG.info("EI Http Reponse Status Code: " + eiResponse.getStatusLine().getStatusCode()
-                + "\nEI Recevied jsonContent:\n" + jsonContent + "\nForwarding response back to EI Frontend WebUI.");
-
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Why removing the HTTP Return code from debug Message? Good to know when debugging the web-server.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java [hunk 5]
```diff
@@ -180,39 +162,11 @@ public String getEIBackendSubscriptionAddress() {
...
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Same question here.

File: src/main/resources/static/js/main.js [hunk 6]
```diff
@@ -5,61 +5,45 @@ jQuery(document).ready(function() {
     var eiffelDocumentationUrlLinks = $('#eiffelDocumentationUrlLinks').text();
 	var frontendServiceUrl = $('#frontendServiceUrl').text();
 
-	document.getElementById("testRulesBtn").onclick = function() {		  
+    function loadMainPage() {
+    	$("#mainFrame").load("subscriptionpage.html");
+	}
 
+	$("#testRulesBtn").click(function() {
     	$("#mainFrame").load("testRules.html");
```
Comment (tobiasake) [inline src/main/resources/static/js/main.js]: Fix indentation.

File: src/main/resources/static/js/main.js [hunk 7]
```diff
@@ -5,61 +5,45 @@ jQuery(document).ready(function() {
...
-	}
+	});
 	
-	document.getElementById("eiInfoBtn").onclick = function() {		  
-  
+	$("#eiInfoBtn").click(function() {
     	$("#mainFrame").load("eiInfo.html");
```
Comment (tobiasake) [inline src/main/resources/static/js/main.js]: Fix indentation.

File: src/main/resources/static/js/main.js [hunk 8]
```diff
@@ -5,61 +5,45 @@ jQuery(document).ready(function() {
...
+		        $("#logoutBlock").hide();
+		        loadMainPage();
+		    }
+		});
+	});
 	
-	document.getElementById("subscriptionBtn").onclick = function() {
-		
+	$("#subscriptionBtn").click(function() {
     	loadMainPage();
```
Comment (tobiasake) [inline src/main/resources/static/js/main.js]: Fix indentation.

File: src/main/resources/static/js/main.js [hunk 9]
```diff
@@ -5,61 +5,45 @@ jQuery(document).ready(function() {
...
-	}
-
-	function loadJmesPathRulesSetUpPage() {
+	});
 
+	$("#jmesPathRulesSetUpBtn").click(function() {
     	$("#mainFrame").load("jmesPathRulesSetUp.html");
```
Comment (tobiasake) [inline src/main/resources/static/js/main.js]: Fix indentation.

**Pull request merged**